mathesar/tests/api/test_user_api.py

```diff
@@ -283,6 +283,17 @@ def test_database_role_create(client, user_bob):
     assert response_data['database'] == database.id
 
 
+def test_database_role_create_non_superuser(client_bob, user_bob):
```

Comment:
The commits that contain the test cases were rejected due to conflict with the upstream. I pushed them now, can you review again and see if any additional test cases should be added.

Reply:
Mostly looks good. It would ideal to add a tests for a scenario where user X had a viewer role on database A, manager role on database B, and no role on database C. We can then test that they can only see database roles for A & B (not C), and they can only create roles for B (and not A or C).

I'm not sure about the status of multiple database tests right now, but if it's easy to do, that would be good to add.